### PR TITLE
chore: replace ESLint plugin node with n

### PR DESCRIPTION
**Pull request by nschonni**
The `eslint-plugin-node` stopped getting updates a little while ago, so a fork was taken into the ESLint Community or to continue support https://github.com/eslint-community/eslint-plugin-n

### CODE REVIEW

File: formatter-pretty/package.json [hunk 1]
```diff
@@ -26,5 +26,8 @@
   },
   "keywords": [
     "markdownlint-cli2-formatter"
-  ]
+  ],
+  "engines": {
+    "node": "^12.22.0 || ^14.17.0 || >=16.0.0"
```
Comment (nschonni) [inline formatter-pretty/package.json:31]: This is required for the Chalk 5 dependency, but it also then let a few disable statements get striped out

File: webworker/webworker-test.js [hunk 2]
```diff
@@ -22,8 +25,6 @@ const outputFormatterLengthIs = (assert, length) => (options) => {
   assert.equal(Object.keys(results).length, length);
 };
 
-/* eslint-disable no-undef */
```
Comment (nschonni) [inline webworker/webworker-test.js:25]: This change isn't required as part of this, but I was looking at some of the disabled rules and though the hoisted versions were a little more explicit

File: .eslintrc.json [hunk 3]
```diff
@@ -8,24 +8,19 @@
     "es6": true
   },
   "plugins": [
-    "node",
+    "n",
     "unicorn"
   ],
   "extends": [
     "eslint:all",
+    "plugin:n/recommended",
```
Comment (nschonni) [inline .eslintrc.json:16]: There is no "all" for this plugin, but in addition to the explicit rules below, I thought opting into the `recommended` would be helpful for any future updates

**Comment by DavidAnson** [comment]
Awesome! I'll have a look tonight.

File: formatter-pretty/package.json (hunk 1 above)
Comment (DavidAnson) [reply formatter-pretty/package.json:31]: Per https://github.com/DavidAnson/markdownlint-cli2/issues/72#issuecomment-1283347887, it looks like I'll be changing CLI2 to be `>=14.18.0`. Can we use the same thing here for consistency?

File: webworker/webworker-test.js (hunk 2 above)
Comment (DavidAnson) [reply webworker/webworker-test.js:25]: I like it!

File: .eslintrc.json (hunk 3 above)
Comment (DavidAnson) [reply .eslintrc.json:16]: Good idea!

File: .eslintrc.json [hunk 4]
```diff
@@ -8,24 +8,19 @@
...
     "plugin:unicorn/all"
   ],
   "ignorePatterns": [
     "test/*/**",
     "webworker/markdownlint-cli2-webworker.js"
   ],
   "reportUnusedDisableDirectives": true,
-  "overrides": [
```
Comment (DavidAnson) [inline .eslintrc.json:23]: Glad to lose this.

File: .eslintrc.json [hunk 5]
```diff
@@ -56,55 +51,55 @@
...
-    "node/no-unpublished-require": "off",
-    "node/no-unsupported-features/es-builtins": "error",
-    "node/no-unsupported-features/es-syntax": "error",
-    "node/no-unsupported-features/node-builtins": ["error", {
+    "n/no-new-require": "error",
+    "n/no-path-concat": "error",
+    "n/no-process-exit": "error",
+    "n/no-unpublished-bin": "error",
+    "n/no-unpublished-import": "error",
+    "n/no-unpublished-require": "error",
```
Comment (DavidAnson) [inline .eslintrc.json:74]: Nice.

File: markdownlint-cli2.js [hunk 6]
```diff
@@ -79,8 +78,8 @@ const importOrRequireResolve = async (dir, id) => {
     try {
       const fileUrlString =
         pathToFileURL(path.resolve(dir, expandId)).toString();
-      // eslint-disable-next-line max-len
-      // eslint-disable-next-line no-inline-comments, node/no-unsupported-features/es-syntax
+
```
Comment (DavidAnson) [inline markdownlint-cli2.js:81]: Extreme nitpick: Remove new blank line?

Comment (DavidAnson) [reply markdownlint-cli2.js:81]: Same for line 448.

File: test/markdownlint-cli2-test-cases.js [hunk 7]
```diff
@@ -87,16 +87,13 @@ const testCases =
   };
 
   const directoryName = (dir) => `${dir}-copy-${host}`;
-
```
Comment (DavidAnson) [inline test/markdownlint-cli2-test-cases.js:90]: Nitpicking again: Restore missing blank line? :)

**Review by DavidAnson — CHANGES_REQUESTED** [review]
Love this, only tiny comments. I'm going to update the engines field now...

File: .eslintrc.json [hunk 8]
```diff
@@ -56,55 +51,55 @@
...
+    "n/no-unsupported-features/es-builtins": "error",
+    "n/no-unsupported-features/es-syntax": "error",
+    "n/no-unsupported-features/node-builtins": ["error", {
       "ignores": [
         "fs.promises"
```
Comment (DavidAnson) [inline .eslintrc.json:79]: ONLY if it's as simple for you as deleting the "ignores" section, is it still necessary to customize this rule, or is `["error"]` enough with the latest `package.json` version changes?

**Comment by nschonni** [comment]
Gave it a rebase to restore the blank lines, and removed the extra config on the remaining rules so they're all just "error" except `n/global-require`

**Review by DavidAnson — APPROVED** [review]
Love it!

**Pull request merged**